src/plugins/ucx/ucx_backend.cpp

```diff
@@ -494,14 +531,14 @@ nixl_status_t nixlUcxEngine::endConn(const std::string &remote_agent) {
 
     nixlUcxConnection &conn = remoteConnMap[remote_agent];
 
-    if(uw->disconnect_nb(conn.ep) < 0) {
-        return NIXL_ERR_BACKEND;
-    }
-
+    bool error = false;
+    for (size_t i = 0; i < conn.eps.size(); i++)
+        if(getWorker(i)->disconnect_nb(conn.getEp(i)) < 0)
```

Comment:
I wonder if its a good idea to have disconnect as an EP routine.

Reply:
WDYM?